src/api/dataops/queries/baseQuery.ts

```diff
@@ -68,6 +68,10 @@ export abstract class BaseQuery<T> {
   private get compiledRequest(): ICompiledRequest<T> {
     const compiledQuery = this.query.compile();
 
+    if (this.records && !Object.keys(compiledQuery).length) {
+      return Object.assign(this.records);
+    }
+
```

Comment:
I do not understand this piece of code. You are returning just `this.recods`

Reply:
I did this validation because for Registers should only send the record itself, unlike other commands that seem to also need the field of action and an array of records (I do not know if this is still in use, so I kept it compatible)